nifi-nar-bundles/nifi-oauth-bundle/nifi-oauth/src/main/java/org/apache/nifi/oauth/AbstractOAuthControllerService.java

```diff
@@ -0,0 +1,172 @@
+    protected Map<String, String> extraHeaders = new HashMap<String, String>();
+    protected String authUrl = null;
+    protected long expireTimeSafetyNetSeconds = -1;
+    protected String accessTokenRespName = null;
+    protected String expireTimeRespName = null;
+    protected String expireInRespName = null;
+    protected String tokenTypeRespName = null;
+    protected String scopeRespName = null;
+
+    public static final PropertyDescriptor AUTH_SERVER_URL = new PropertyDescriptor
+            .Builder().name("OAuth2 Authorization Server URL")
+            .displayName("OAuth2 Authorization Server")
+            .description("OAuth2 Authorization Server that grants access to the protected resources on the behalf of the resource owner.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_ACCESS_TOKEN_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'access_token' name")
+            .displayName("JSON response 'access_token' name")
+            .description("Name of the field in the JSON response that contains the access token. IETF OAuth2 spec default is 'access_token' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("access_token")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_EXPIRE_TIME_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'expire_time' name")
+            .displayName("JSON response 'expire_time' name")
+            .description("Name of the field in the JSON response that contains the expire time. IETF OAuth2 spec default is 'expire_time' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("expire_time")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_EXPIRE_IN_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'expire_in' name")
+            .displayName("JSON response 'expire_in' name")
+            .description("Name of the field in the JSON response that contains the expire in. IETF OAuth2 spec default is 'expire_in' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("expire_in")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_TOKEN_TYPE_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'token_type' name")
```

Comment:
The `name` should avoid special characters and spaces for serialization purposes. Suggest `.name("JSON_response_token_type_name")`.